File: 动态规划/一维子数组、子序列(定义状态为以nums[i]结尾的XXXX)/674. 最长连续递增序列.js
/*
 * dp[i] 表示：以 nums[i] 结尾 的「连续递增的子数组」的长度。注意：这个定义中 nums[i] 必须被选取，且必须是这个子数组的最后一个元素；
 *
 * 与 300 的区别就在于，这道题求的是 subarray，而 subarray 是不能通过删减nums中的元素得到的，所以在dp中递推时，只用看 dp[i-1] 而不像 300 要看 dp[j]
 */
var findLengthOfLCIS = function (nums) {
    let dp = new Array(nums.length).fill(1);
    let result = 1;
    for (let i = 1; i < nums.length; i++) {
        if (nums[i] > nums[i - 1]) dp[i] = dp[i - 1] + 1;
        result = Math.max(result, dp[i]);
    }
    // console.log(dp);
    return result;
};
findLengthOfLCIS([1, 3, 5, 4, 7]);

File: 数组/滑动窗口/计数问题(最大窗口)/674. 最长连续递增序列.py

#> 这道题也能用动态规划解决 
class Solution(object):
    def findLengthOfLCIS(self, nums):
        maxLen = 0
        
        left = 0
        right = 0
        
        while right < len(nums):
            
            if right > 0 and nums[right] > nums[right - 1]:
                maxLen = max(maxLen, right - left + 1)
                
            #* 如果 nums[right - 1]『大于等于』nums[right], 则不满足连续递增的 subsequence 的条件, 缩小滑动窗口, 使得左指针右指针重合, 代表此时连续递增子序列就是当前字符 1 个, 长度为 1 
            if right > 0 and nums[right] <= nums[right - 1]:
                left = right
            
            right += 1        
        return maxLen if maxLen != 0 else 1
